luatest/server.lua

```diff
@@ -431,6 +433,32 @@ local function is_header_set(headers, name)
     return false
 end
 
+-- Checking for the correctness of the passed `args` for the `fn` function
+local function is_malformed_args(fn, args)
+    local fn_nparams = debug.getinfo(fn).nparams
+    local fn_source = string.split(debug.getinfo(fn).source, '/')
+    local fn_details = string.format(
+        '%s:%s', fn_source[#fn_source], debug.getinfo(fn).linedefined) -- 'filename:line'
+
+    if args == nil then
+        if fn_nparams > 0 then
+            return string.format(MALFORMED_MSG, fn_details, fn_nparams, 'nil')
+        end
+    else
+        if #args ~= fn_nparams then
```

Comment:
`#args` is bad at determining real table size, it counts only the "array" part of the table.

```lua
tarantool> args = {1,2,3, [7] = 4}
---
...

tarantool> args
---
- - 1
  - 2
  - 3
  - null
  - null
  - null
  - 4
...

tarantool> #args
---
- 3
...

```

But I think this is ok for our purpose. It would probably be an overkill to detect cases like that.

Reply:
@Totktonada hi!
 Is there a more convenient way to work with the number of items in the table?

Reply:
I'm out of context here, but if you need to count elements in a table (and don't want to define an utility function), you can use luafun:

```lua
tarantool> fun = require('fun')
tarantool> fun.wrap(pairs({a = 5, b = 4, 1, 2, 3})):length()
---
- 5
...
```